pybites_tools/alarm.py

```diff
@@ -0,0 +1,41 @@
+import argparse
+import os
+import time
+
+from dotenv import load_dotenv
+from playsound import playsound
+
+load_dotenv()
+
+ALARM_MUSIC_FILE = os.environ["ALARM_MUSIC_FILE"]
+
+
+def countdown(seconds: int) -> None:
+    while seconds:
+        mins, secs = divmod(seconds, 60)
+        print(f"{mins:02}:{secs:02}", end="\r")
+        time.sleep(1)
+        seconds -= 1
+
+    print("00:00", end="\r")
+    playsound(ALARM_MUSIC_FILE)
+
+
+def main():
+    parser = argparse.ArgumentParser("Play an alarm after N minutes")
+    parser.add_argument(
+        "-m",
+        "--minutes",
+        default=25,
+        help="Number of minutes to countdown from (default 25 = Pomodoro)",
+    )
+    args = parser.parse_args()
+    seconds = int(args.minutes) * 60
+    try:
+        countdown(seconds)
+    except KeyboardInterrupt:
+        pass
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
This requires the script to run in foreground right? So maybe we should change this to a daemon script that just starts the main script as a subprocess like a real alarm so you can do something else in the console while the alarm is counting down in the backgorund

Reply:
I am not sure you can do that natively with Python without the use of external tools like a task queue? On the other hand it's easy enough to add an ampersand in Unix to push the process to the background.

Reply:
Ah ok again Linux is amazing

Reply:
Indeed! Could do this with `os.system` maybe

Reply:
OK this is what I came up with:

```
(venv) √ pybites-tools (alarm) $ python -m pybites_tools.alarm -m 1 -b
Playing alarm in 1 minute
(venv) √ pybites-tools (alarm) $ python -m pybites_tools.alarm -m 2 -b
Playing alarm in 2 minutes

$ ps -ef |grep alarm
  501 71847     1   0  1:41PM ttys002    0:00.31 /Library/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python -m pybites_tools.alarm -m 1
  501 71857     1   0  1:41PM ttys002    0:00.31 /Library/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python -m pybites_tools.alarm -m 2
  501 71980 69986   0  1:42PM ttys002    0:00.00 grep alarm
```

https://github.com/PyBites-Open-Source/pybites-tools/pull/3/commits/2a437aa65ccc10a57df38b45a71453fb923fa6f3